packages/core/src/utils/getFieldType.ts

```diff
@@ -13,7 +13,14 @@ const getFieldType = (
   schema: FormupYupSchema,
   options?: getSchemaFieldOptions,
 ) => {
-  const schemaField = getSchemaField(name, schema, options);
+  const defaultOptions = {
+    returnSubtype: true,
+  };
+
+  const schemaField = getSchemaField(name, schema, {
+    ...defaultOptions,
+    ...options,
```

Comment:
If options is null this would break

```suggestion
    ...(options || {}),
```

Reply:
Agreed.